### PR TITLE
feat: reintroduce linked flag variant to decide

### PR DESCRIPTION
**Pull request by pauldambra**
see:

* https://github.com/PostHog/posthog/pull/20624
* https://github.com/PostHog/posthog/pull/20543
* https://github.com/PostHog/posthog-js/pull/1040

We introduced the ability to link a flag variant not just a flag for controlling session replay
This was the last deploy before a decide incident
We can't see why it would have caused an incident
Which is going to make it tricky to safely reintroduce 🙈 

What makes this (maybe) safe?

* I've moved the entire session recording config response generation into its own function, wrapped in a catch, and defaulting to False
* I'm not re-using the linked_flag variable... maybe we were leaking an unexpected shape response causing clients to error and retry leading to unexpected load

### CODE REVIEW

File: posthog/api/decide.py [hunk 1]
```diff
@@ -93,11 +93,10 @@ def get_decide(request: HttpRequest):
         "isAuthenticated": False,
         # gzip and gzip-js are aliases for the same compression algorithm
         "supportedCompression": ["gzip", "gzip-js"],
+        "featureFlags": [],
+        "sessionRecording": False,
     }
 
-    response["featureFlags"] = []
-    response["sessionRecording"] = False
```
Comment (pauldambra) [inline posthog/api/decide.py:99]: fly-by I don't _think_ there's a reason not to declare then in with the dict creation

File: posthog/api/decide.py [hunk 2]
```diff
@@ -336,3 +292,53 @@ def get_decide(request: HttpRequest):
...
+                session_recording_config_response.update(
+                    {
+                        "recordCanvas": record_canvas,
+                        # hard coded during beta while we decide on sensible values
+                        "canvasFps": 4 if record_canvas else None,
+                        "canvasQuality": "0.6" if record_canvas else None,
+                    }
+                )
+    except Exception as e:
+        capture_exception(e)  # we don't want to fail decide if session recording config fails to load
```
Comment (pauldambra) [inline posthog/api/decide.py:342]: swallow but report any error

File: posthog/api/decide.py [hunk 3]
```diff
@@ -336,3 +292,53 @@ def get_decide(request: HttpRequest):
...
+
+    return session_recording_config_response
```
Comment (pauldambra) [inline posthog/api/decide.py:344]: this defaults to False so in the case of an error we turn off session replay, that's not great but better than breaking decide

**Review by neilkakkar — APPROVED** [review]
Seems reasonable 👍

**Pull request merged**